src/Featurizers/Components/DocumentStatisticsEstimator.h

```diff
@@ -17,31 +21,6 @@ namespace Components {
 
 namespace {
 
-//this is temporary split algorithm
-inline void split_temp(std::string const & input, std::function<void (std::string::const_iterator&, std::string::const_iterator&)> const & callback) {
-
-    std::string::const_iterator left = input.begin();
-    std::string::const_iterator right = left;
-
-    while (left != input.end() && *left == ' ')
-        ++left;
-
-    right = left;
-    while (right != input.end()) {
-        if (*right == ' ') {
-            callback(left, right);
-            left = right;
-            while (left != input.end() && *left == ' ')
-                ++left;
-            right = left;
-        } else {
-            ++right;
-        }
-    }
-    if (left != right)
-        callback(left, right);
-}
-
 struct IterRangeComp {
```

Comment:
>struct IterRangeComp [](start = 0, length = 20)

This appears to be an implementation detail - move its definition to the Implementation section of the code.